FLIR/conservator/wrappers/media.py

```diff
@@ -204,28 +209,34 @@ def upload(conservator, upload_request):
         """
         assert isinstance(upload_request, MediaUploadRequest)
         file_path = upload_request.file_path
+        original_name = os.path.split(file_path)[-1]
         remote_name = upload_request.remote_name
         collection_id = upload_request.collection_id or None
 
         file_path = os.path.expanduser(file_path)
         assert os.path.isfile(file_path)
         if remote_name is None:
-            remote_name = os.path.split(file_path)[-1]
+            remote_name = original_name
 
         media = None
         try:
-            media = MediaType._create(conservator, remote_name, collection_id)
-            upload_id = media._initiate_upload(remote_name)
+            media = MediaType._create(conservator, original_name, collection_id)
+            upload_id = media._initiate_upload(original_name)
 
             url = media._generate_signed_upload_url(upload_id)
             response = conservator.files.upload(url=url, local_path=file_path)
             completion_tag = response.headers["ETag"]
 
             media._complete_upload(
-                remote_name, upload_id, completion_tags=[completion_tag]
+                original_name, upload_id, completion_tags=[completion_tag]
             )
             media._trigger_processing()
 
+            if original_name != remote_name:
```

Comment:
This call can, and should, be made on line 224 above. If the file being uploaded is a simple image file, it may already be processed by the time this call is made, and thus this mutation would fail. Also, if `remote_name` is `None` (or `""`, which seems to be the default value if not supplied), then `name` should not be set.

Reply:
Ok, moved the block and checked that remote_name is set.